### PR TITLE
convert jboss-logmanager-mdc groovy test to java

### PR DESCRIPTION
**Pull request by cuichenli**
the `getMdcCopy` test is omitted as it is only available in jboss-logmanager 1.3+ and the test depends on jboss-logmanager 1.1

### CODE REVIEW

File: instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/groovy/JbossLogmanagerMdcTest.groovy [hunk 1]
```diff
@@ -1,113 +0,0 @@
...
-      logger.info("log message 3")
-      Span.current()
-    }
-
-    then:
-
-    logRecords.size() == 3
-
-    // The method getMdcCopy exists only in jboss-logmanager 1.3+
-    def hasGetMdcCopy = logRecords.get(0).metaClass.getMetaMethod("getMdcCopy") != null
```
Comment (mateuszrzeszutek) [inline instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/groovy/JbossLogmanagerMdcTest.groovy:81]: WDYT about using reflection to get this? It'd be nice to preserve this verification when running against the latest version (`latestDepTest`)

**Review by mateuszrzeszutek — COMMENTED** [review]
Thanks for help @cuichenli !

File: instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/groovy/JbossLogmanagerMdcTest.groovy (hunk 1 above)
Comment (cuichenli) [reply instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/groovy/JbossLogmanagerMdcTest.groovy:81]: updated, please check, thanks!

File: instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/mdc/v1_1/JbossLogmanagerMdcTest.java [hunk 2]
```diff
@@ -90,11 +93,28 @@ void idsGeneratedWhenSpanProvided() throws NoSuchMethodException {
 
     assertThat(logRecords.size()).isEqualTo(3);
 
+    Method getMdcCopy = null;
+    boolean hasGetMdcCopy = false;
+    try {
+      getMdcCopy = logRecords.get(0).getClass().getMethod("getMdcCopy");
+      hasGetMdcCopy = getMdcCopy != null;
```
Comment (mateuszrzeszutek) [inline instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/mdc/v1_1/JbossLogmanagerMdcTest.java:100]: `getMdcCopy != null` is always true at this point - consider removing the boolean flag and just checking for `getMdcCopy != null` in the conditions below.

File: instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/build.gradle.kts [hunk 3]
```diff
@@ -13,4 +13,5 @@ muzzle {
 
 dependencies {
   library("org.jboss.logmanager:jboss-logmanager:1.1.0.GA")
+  latestDepTestLibrary("org.jboss.logmanager:jboss-logmanager:+")
```
Comment (mateuszrzeszutek) [inline instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/build.gradle.kts:16]: This actually is added automatically, there's no need to specify it manually (unless you want to limit the latest version; which is not the case here)
```suggestion
```

File: instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/mdc/v1_1/JbossLogmanagerMdcTest.java (hunk 2 above)
Comment (cuichenli) [reply instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/mdc/v1_1/JbossLogmanagerMdcTest.java]: updated, thanks!

**Comment by cuichenli** [comment]
the test failed on another module, looks like i do not have the permission to re-run it.

**Comment by laurit** [comment]
@cuichenli you don't need to worry about test-latest-deps failure we'll fix it and let you know if this pr needs to be rebased.

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
@opentelemetrybot update

**Comment by mateuszrzeszutek** [comment]
(closing & reopening to trigger checks)

**Pull request merged**